mmcv/fileio/file_client.py

```diff
@@ -411,6 +412,436 @@ def _list_dir_or_file(dir_path, list_dir, list_file, suffix,
                                  recursive)
 
 
+class AWSBackend(BaseStorageBackend):
+    """AWSBackend is Amazon Simple Storage Service(s3).
+
+    AWSBackend supports reading and writing data to aws s3.
+    It relies on awscli and boto3, you must install them and run
+    ``aws configure`` in advance to use it.
+
+    .. code-block:: shell
+        :linenos:
+
+        pip install awscli
+        pip install boto3
+
+        # configure your aws credentials
+        # Note that region must match your bucket region.
+        aws configure
+        # AWS Access Key ID [****************JQGK]:
+        # AWS Secret Access Key [****************3gUp]:
+        # Default region name [ap-east-1]:
+        # Default output format [json]:
+
+    Note:
+        In v1.4.7 and later, AWSBackend is provided to support reading and
+        writing data to s3.
+
+    Args:
+        path_mapping (dict, optional): Path mapping dict from local path to
+            Petrel path. When ``path_mapping={'src': 'dst'}``, ``src`` in
+            ``filepath`` will be replaced by ``dst``. Default: None.
+
+    Examples:
+        >>> from mmcv.fileio.file_client import AWSBackend
+        >>> filepath = 's3://bucket/obj'
+        >>> client = AWSBackend()
+        >>> client.get(filepath1)  # get data from aws s3
+        >>> client.put(obj, filepath)
+    """
+
+    def __init__(self, path_mapping: Optional[dict] = None):
+        try:
+            import boto3
+            import boto3.s3.transfer
+            import botocore.exceptions
+        except ImportError:
+            raise ImportError('Please install boto3 to enable AWSBackend '
+                              'by "pip install boto3".')
+
+        self._client = boto3.client('s3')
```

Comment:
boto3 client is not serializable. When using it with multiprocessing, it will break

Reply:
Like this? cc: @274869388 

```
@property
def client(self):
    if self._client is None:
        self._client = boto3.client('s3')
    return self._client
```